{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Align reads using bowtie. Filter only reads that aligned and convert the output to a bedfile\n",
    "##### Bedfile will have the positions of every read that aligned to given reference genome (in this case NC_000913.2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "!bowtie -S -m 1 -v 0 indexes/NC_000913.2 netseq_data/clean_Trimmed_wt-mmc.fastq.gz 2> wt_mmc_NET.log|samtools view -b -h -F 4|bedtools bamtobed -i '-' > wt_mmc_NET_aligned_2.bed "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Split alignments by strand and edit the alignment bedfile so that it only keeps the 3' end of each gene\n",
    "##### 'plus_3end.bed' and 'minus_3end.bed' are the outputfiles that now only have the position of the 3' base"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "!grep -w '+' wt_mmc_NET_aligned_2.bed|awk '{print $1\"\\t\"$2\"\\t\"($2+1)\"\\t\"\".\"\"\\t\"$5\"\\t\"$6}'> plus_3end_2.bed\n",
    "!grep -w '-' wt_mmc_NET_aligned_2.bed|awk '{print $1\"\\t\"($3-1)\"\\t\"($3)\"\\t\"\".\"\"\\t\"$5\"\\t\"$6}'> minus_3end_2.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Import packages needed for plotting"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "%matplotlib inline  \n",
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Make a pandas dataframe of each positon of the E. coli genome (index) and the corresponding 3' base counts that are present in that position."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "p_3end=pd.read_table('plus_3end_2.bed',header=None)\n",
    "m_3end=pd.read_table('minus_3end_2.bed',header=None)\n",
    "p_counts=pd.DataFrame(p_3end[2].value_counts()).sort_index().reindex(range(1,4641653),fill_value=0)\n",
    "m_counts=pd.DataFrame(m_3end[2].value_counts()).sort_index().reindex(range(1,4641653),fill_value=0)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Import bedfile so that gene positions can be determined (in future will also make GFF/GTF file an option)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_annot=pd.read_csv(\"NC_000913.2.bed\",sep='\\t',header=None)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Find gene of interest in the befile and store the coordinates of the gene location (in the future will make it so a list of genes can be given)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene='thrL'\n",
    "gene_strand=gene_annot[gene_annot[4]==gene].iloc[0][5]\n",
    "gene_pos=((gene_annot[gene_annot[4]==gene].iloc[0][1])-12,((gene_annot[gene_annot[4]==gene].iloc[0][2])+13))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Only keep counts that reflect transcription happening on the sense strand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "if gene_strand == '+':\n",
    "    gene_plot=m_counts.iloc[gene_pos[0]:gene_pos[1],:]\n",
    "if gene_strand == '-':\n",
    "    gene_plot=p_counts.iloc[gene_pos[0]:gene_pos[1],:]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Plot the counts in a barplot (reverse the orientation of the plot if gene is '-'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA3AAAACFCAYAAAAaXRrgAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDIuMi4yLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvhp/UCwAACGlJREFUeJzt3U+InGcdB/Dvr0mrEtCoxFCSYFsMjafWGmq9SatYvdSDhV60yEgu2rtXwYOeSgURQ1cpItbSS4OIUqq9xqb0D7S1NIZqlto0YlsQUaw8HvatLukm+6Y7MzvPzucDy7zv+zw77zPv7Lw73/m97zvVWgsAAACL74rtHgAAAADjCHAAAACdEOAAAAA6IcABAAB0QoADAADohAAHAADQiVEBrqr2VtXDVfWHqnqhqj5dVR+qqker6qXh9oND36qq71fV6ap6tqpumu1DAAAAWA5jK3D3Jfl1a+1IkhuSvJDkW0kea60dTvLYMJ8kX0hyePg5luSHUx0xAADAkqrNvsi7qt6f5Jkk17V1navqxSSfaa39paquTvJ4a+36qvrRMP3zC/vN7FEAAAAsgTEVuOuSnE/yk6p6qqrur6o9Sfa/HcqG248M/Q8kObvu91eHZQAAAGzB7pF9bkpyT2vtZFXdl/8fLrmR2mDZO8p8VXUsa4dYZs+ePZ88cuTIiKEAAADT8PLL/75o2zXXXDnHkZAkTz755F9ba/s26zcmwK0mWW2tnRzmH85agDtXVVevO4TytXX9D637/YNJXrnwTltrx5McT5KjR4+2U6dOjRgKAAAwDZPJuYu2razsn+NISJKq+tOYfpseQtlaezXJ2aq6flh0W5Lnk5xIcvew7O4kjwzTJ5J8dbga5S1J3nT+GwAAwNaNqcAlyT1JflZVVyU5k+RrWQt/D1XVJMmfk9w59P1Vki8mOZ3kH0NfAAAAtmhUgGutPZ3k6AZNt23QtyX5xhbHBQAAwAXGfg8cAAAA20yAAwAA6IQABwAA0AkBDgAAoBMCHAAAQCcEOAAAgE4IcAAAAJ0Q4AAAADohwAEAAHRCgAMAAOiEAAcAANAJAQ4AAKATAhwAAEAnBDgAAIBOCHAAAACdEOAAAAA6IcABAAB0QoADAADohAAHAADQCQEOAACgEwIcAABAJwQ4AACATghwAAAAnRDgAAAAOiHAAQAAdGL3dg8AAACSZDI5d9G2lZX9cxwJLC4VOAAAgE4IcAAAAJ0Q4AAAADohwAEAAHRCgAMAAOiEAAcAANAJAQ4AAKATAhwAAEAnBDgAAIBOCHAAAACd2L3dAwAAAMaZTM5dsn1lZf+cRsJ2UYEDAADohAAHAADQCQEOAACgEwIcAABAJwQ4AACATghwAAAAnRDgAAAAOiHAAQAAdEKAAwAA6MSmAa6q3ltVv6+qZ6rquar69rD82qo6WVUvVdUvquqqYfl7hvnTQ/s1s30IAAAAy2FMBe5fSW5trd2Q5MYkt1fVLUm+l+Te1trhJK8nmQz9J0leb619LMm9Qz8AAAC2aNMA19b8fZi9cvhpSW5N8vCw/IEkXxqm7xjmM7TfVlU1tREDAAAsqVHnwFXVrqp6OslrSR5N8sckb7TW3hq6rCY5MEwfSHI2SYb2N5N8eIP7PFZVp6rq1Pnz57f2KAAAAJbAqADXWvtPa+3GJAeT3Jzk4xt1G243qra1dyxo7Xhr7Whr7ei+ffvGjhcAAGBpXdZVKFtrbyR5PMktSfZW1e6h6WCSV4bp1SSHkmRo/0CSv01jsAAAAMtszFUo91XV3mH6fUk+m+SFJL9L8uWh291JHhmmTwzzGdp/21p7RwUOAACAy7N78y65OskDVbUra4HvodbaL6vq+SQPVtV3kjyVZGXov5Lkp1V1OmuVt7tmMG4AAICls2mAa609m+QTGyw/k7Xz4S5c/s8kd05ldAAAAPzPZZ0DBwAAwPYR4AAAADohwAEAAHRCgAMAAOiEAAcAANAJAQ4AAKATAhwAAEAnxnyRNwBbNJmcu2jbysr+OY4EAOiZChwAAEAnBDgAAIBOCHAAAACdEOAAAAA6IcABAAB0QoADAADohAAHAADQCQEOAACgEwIcAABAJ3Zv9wAAYFFNJucu2raysn+OIwGANSpwAAAAnRDgAAAAOuEQSrgEh08BALBIVOAAAAA6IcABAAB0QoADAADohAAHAADQCQEOAACgE65CCSwMV/0EALg0FTgAAIBOCHAAAACdcAglAADMmNMEmBYVOAAAgE6owAEA8K5dqrKUqC7BtKnAAQAAdEKAAwAA6IRDKAEAYAE4HJUxVOAAAAA6oQIHwJa4NDaLRhWDebMfZJ4EOObCjg3mx5tXYBF5LwDTIcABAADvmg8O58s5cAAAAJ1QgQMAgItw6OfO1WvlUICDjvS6owEWj/0J7GzLHjx38uMX4AAAWDo+xFhMOzl4TYsABwAAdGPZQ54Ax1Lyqds4y76D3Ol6fX57HTcA/Znme8Zp/f+aSYCrqtuT3JdkV5L7W2vfncV6oBfecDKGv5PpGbMtbW+AzdlXLp6pB7iq2pXkB0k+l2Q1yRNVdaK19vy018VsLWKVyk5knHluJ8+J7b3sxuwrx+5Pd3Lw7HXci8i2hOU2iwrczUlOt9bOJElVPZjkjiRLG+AWMQjtZLb34lnEww+8AVo8Xrv98rpk0difLJ6d/pzMc/81iwB3IMnZdfOrST41g/XM3KJ8qn656+v1E9xFHNMY83jjcrn3tdX1LfL2HqPX52Qn28l/bzvZIr4GFvH/5Vb7vN2vt8d2uWOaFvsTll211qZ7h1V3Jvl8a+3rw/xXktzcWrvngn7HkhwbZq9P8uJUBwIAANCPj7bW9m3WaRYVuNUkh9bNH0zyyoWdWmvHkxyfwfoBAAB2pCtmcJ9PJDlcVddW1VVJ7kpyYgbrAQAAWCpTr8C11t6qqm8m+U3Wvkbgx62156a9HgAAgGUz9XPgAAAAmI1ZHEIJAADADAhwAAAAnRDgAAAAOiHAAQAAdEKAAwAA6IQABwAA0AkBDgAAoBMCHAAAQCf+C+3vL1HfL7e9AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 1080x144 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "x=gene_plot.index.tolist()\n",
    "y=gene_plot[2].tolist()\n",
    "if gene_strand=='-':\n",
    "    y=y[::-1]\n",
    "plt.figure(figsize=(15,2))\n",
    "sns.barplot(x=x,y=y,color='blue',alpha=0.7)\n",
    "sns.despine(top=False,right=False,bottom=False,left=False)\n",
    "\n",
    "\n",
    "plt.xticks(np.arange(min(x), max(x)+1, 100))\n",
    "plt.ylim(0,600)\n",
    "plt.yticks([0,300,600])\n",
    "plt.xticks([])\n",
    "#plt.savefig('thrL_netseq_example.pdf', bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Figure size 432x288 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.savefig('thrL_netseq_example.pdf', bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [default]",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
